runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -698,23 +701,23 @@ public TimerInternalsTimer(
 
     @Override
     public void set(Instant target) {
+      this.target = target;
```

Comment:
It would be nice that the `IllegalArgumentException` is thrown before the wrong value is set. Here it is not likely to cause a problem, but it is a good practice. You never know how this class might be used in the future.

Reply:
what would be a considered as illegal deliveryTimestamp?

Reply:
Same as would be thrown in `setAndVerify` - and as an aside, I'd much prefer the timestamp be returned by verify and the fields be final if that's possible (I think it is)